arm-software/shared/elf2bin/elf.cpp

```diff
@@ -0,0 +1,69 @@
+// Code to read ELF data structures for elf2bin
+//
+// Copyright (c) 2022-2025, Arm Limited and affiliates.
+//
+// Part of the Arm Toolchain project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+
+#include "elf2bin.h"
+
+using namespace llvm;
+using namespace llvm::object;
+
+template <typename ELFT>
+static std::vector<Segment>
+get_segments(InputObject &inobj, ELFObjectFile<ELFT> &elfobj, bool physical) {
+  std::vector<Segment> segments;
+
+  Expected<ArrayRef<typename ELFT::Phdr>> phdrs_or_err =
+      elfobj.getELFFile().program_headers();
+  if (!phdrs_or_err) {
+    fatal(inobj, "unable to read program header table",
```

Comment:
A user might not understand that a relocatable object doesn't have a program header table. Perhaps worth adding a sentence like "elf2bin works on executables or shared libraries."